caddyhttp/httpserver/server.go

```diff
@@ -406,3 +417,14 @@ func WriteTextResponse(w http.ResponseWriter, status int, body string) {
 	w.WriteHeader(status)
 	w.Write([]byte(body))
 }
+
+// HandleCommonError checks if the given error is an error that could pop up
+// when handling request. Currently only checks for MaxBytesReader error.
+// Returns a boolean - whether the given error is a common error, and a int -
+// the suggested HTTP status code to return to the user
+func IsCommonError(err error) (bool, int) {
+	if err.Error() == "http: request body too large" {
```

Comment:
Reading the string content of an error value is a little scary -- I like to avoid this where possible. It could change any time.

After looking into MaxBytesReader, I see it [makes use of an unexported error value](https://github.com/golang/go/blob/89c4cbd7acc36cf627746f379f0cc002b5b60383/src/net/http/request.go#L982), hence the need for this function which has to read the error string...

Finding https://github.com/golang/go/issues/8479, it seems like the Go team isn't interested in having the error value exported and reused, but the implementation of MaxBytesReader is trivial; we could just borrow from it and change it to meet our needs by using a pre-defined error value. Also it would be useful to mention how big the read body was, just so it's easier to read in the logs.

However, we should decide if we want to centralize the use of a MaxBodyReader like this, or if each middleware that reads the response body should use it instead. Either way, they'll have to check for the kind of error value returned and, if an EntityTooLargeError, they return a special status code in that case.